packages/dependencies/src/node-modules.ts

```diff
@@ -14,54 +14,56 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import * as fs from "node:fs"
+import { type Dirent, promises as fs } from "node:fs"
+import { createRequire } from "node:module"
 import * as path from "node:path"
 
+import { callsite } from "@solarwinds-apm/module"
+
 import { type Dependencies, type Package } from "."
 
 export function collectNodeModulesDependencies(dependencies: Dependencies) {
   // node_modules is not a real package we just want all the paths looked up by
   // package resolution
-  const roots = require.resolve.paths("node_modules") ?? []
-  for (const root of roots) {
-    collectRoot(dependencies, root)
-  }
+  const roots =
+    createRequire(callsite().getFileName()!).resolve.paths("node_modules") ?? []
+  const tasks = roots.map((root) => collectRoot(dependencies, root))
+  return Promise.allSettled(tasks)
 }
 
-function collectRoot(dependencies: Dependencies, root: string) {
-  let entries: fs.Dirent[]
+async function collectRoot(dependencies: Dependencies, root: string) {
+  let entries: Dirent[]
   try {
-    entries = fs.readdirSync(root, { withFileTypes: true })
+    entries = await fs.readdir(root, { withFileTypes: true })
   } catch {
     return
   }
 
-  for (const entry of entries) {
+  const tasks = entries.map(async (entry) => {
     const entryPath = path.join(root, entry.name)
-    const stats = fs.statSync(entryPath, { throwIfNoEntry: false })
+    const stats = await fs.stat(entryPath)
+
     // Skip files and hidden directories
-    if (!stats?.isDirectory() || entry.name.startsWith(".")) {
-      continue
+    if (!stats.isDirectory() || entry.name.startsWith(".")) {
+      return
     }
 
     // Scoped packages are nested under a directory named after the scope
     if (entry.name.startsWith("@")) {
-      collectRoot(dependencies, entryPath)
-      continue
+      await collectRoot(dependencies, entryPath)
+      return
     }
 
-    try {
-      const packagePath = path.join(entryPath, "package.json")
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const { name, version } = require(packagePath) as Package
-      dependencies.add(name, version)
+    const packagePath = path.join(entryPath, "package.json")
```

Comment:
How come this (and similar blocks in other files) was moved out of the try-catch? Will the `packageJson` pretty much always exist now because of dependency on our own instrumentations package?

Reply:
No, and I don't plan on making that possible since that package was causing a huge amount of problems

Reply:
Should probably be in a try-catch actually, woops